modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptClientCodegen.java

```diff
@@ -1147,7 +1147,7 @@ private String toExampleValueRecursive(String modelName, Schema schema, Object o
                     RxGen does not support our ECMA dialect https://github.com/curious-odd-man/RgxGen/issues/56
                     So strip off the leading / and trailing / and turn on ignore case if we have it
                      */
-                    Pattern valueExtractor = Pattern.compile("^/?(.+?)/?(.?)$");
+                    Pattern valueExtractor = Pattern.compile("^/?(.+)?/?(.?)$");
```

Comment:
this regex now does not strip trailing `/`. although I have no context about this code, the preceding comment indicates that this should happen.
However, it might be simpler to use string manipulation instead of the regexp, i.e. 
- strip the leading and trailing slash
- split by slash
- join the results again (so we get two groups, results[0] and results[1...])

wdyt?

Reply:
If I understand it correctly, the leading and trailing slash is `/` in the regex pattern which is already extracted in the next `if`. If the input is `/[A-Z]/`, then it’ll changing to `[A-Z]`. So if it’s, we dont need to do anything about it.

Reply:
Please correct me if I am wrong 🙇‍♂️

Reply:
I still think that regex is still simpler. If we want to change from regex to string, we might want to detect the pattern string beginning/ending with `/`. If it's we might need to check the regex option after `/` to use with `RgxGen` to generate an example with case insensitive. And I don't think we can split because the regex pattern can contain the slash.

I think the way to fix this is making the regex pattern extraction more strictly. Currently, the generator cannot extract the regex option because of `/?` is used (here is a reproducing code https://regex101.com/r/m1mC9Q/1). So we might fix it by removing `?` after the `/`, it'll be more clear that the generator will extract regex option if the user uses a pattern that begins/ends with `/`. 

But since I'm not expert in Java,  I would like to hear your feedback if it has a greater way to fix this issue.

Reply:
@bodograumann @TiFu can you advise here?

Reply:
I cannot find a reference that says, the pattern field should allow this kind of `/regex/i` format in the first place.
Cff.
- https://swagger.io/docs/specification/data-models/data-types/#string
- https://262.ecma-international.org/11.0/#sec-regexp-regular-expression-objects
- https://stackoverflow.com/a/60773255/1534459

Do we have some generated code from the pattern and what is the behaviour there?
Maybe this stripping of `/` should simply be dropped.